acl_loader/main.py

```diff
@@ -422,7 +445,12 @@ def convert_ip(self, table_name, rule_idx, rule):
         # so there isn't currently a good way to check if the user defined proto=0 or not.
         if rule.ip.config.protocol:
             if rule.ip.config.protocol in self.ip_protocol_map:
-                rule_props["IP_PROTOCOL"] = self.ip_protocol_map[rule.ip.config.protocol]
+                # Special case: "1" is the wrong protocol number for ICMPV6, we need to use "58" instead if
```

Comment:
>Special case: "1" is the wrong protocol number for ICMPV6 [](start = 18, length = 57)

Do you mean input acl data could be wrong?
Is it better to exit with error instead of correcting it?

Reply:
No. ICMP has a different protocol number for IPv4 and IPv6. Since user passes a string describing the protocol they want to match, we need to do a conversion. I thought this was slightly cleaner than having an entire separate case for `rule.ip.config.protocol` == "IP_ICMP"